scRNA-seq-advanced/03-dataset_integration.Rmd

```diff
@@ -445,37 +612,68 @@ scater::plotReducedDim(merged_sce,
                        # Some styling to help us see the points:
                        point_size = 0.5,
                        point_alpha = 0.2) +
-  ggplot2::ggtitle("UMAP after integration with fastMNN") +
-  # Modify the ggplot2 guides so that the legend points are larger and easier to see:
-  ggplot2::guides(colour = ggplot2::guide_legend(override.aes = list(size = 3, alpha = 1)))
+  # Modify legend so they key is larger and easier to see
+  guides(colour = guide_legend(override.aes = list(size = 3, alpha = 1))) +
+  # add plot title
+  ggtitle("UMAP after integration with fastMNN")
 ```
 
 
 
+This `fastMNN_UMAP` certainly looks different from the one we made from `merged_UMAP`!
+It particularly looks like `SCPCL000479` and `SCPCL000480` have substantial overlap - intriguingly, these two samples are from the same patient, so this overlap may have some true biological underpinnings!
+
+However, it's a bit challenging to see all the points given the amount of overlap in the plot.
+One way we can all the points a bit better is to facet the plot by sample:
+
+```{r fastmnn umap batches faceted}
+scater::plotReducedDim(merged_sce,
+                       dimred = "fastMNN_UMAP",
+                       colour_by = "sample",
+                       point_size = 0.5,
+                       point_alpha = 0.2, 
+                       # Allow for faceting by a variable using `other_fields`:
+                       other_fields = "sample") +
```

Comment:
Just as a note, I looked into this a bit, and we _could_ leave this off and use `facet_wrap(~colour_by)`, but that is so non-intuitive that I definitely don't want to do it!

Reply:
Oh no...so it actually makes those arguments columns into a df! Great....